test/NuGet.Services.Staging.EndToEndTests/StagingEndToEnd.cs

```diff
@@ -0,0 +1,210 @@
+
+            configurationRoot.GetSection("StagingEndToEnd").Bind(this);
+        }
+    }
+
+    public class StagingEndToEnd
+    {
+        private readonly ITestOutputHelper _output;
+        private readonly StagingEndToEndConfiguration _configuration;
+        
+        public StagingEndToEnd(ITestOutputHelper output)
+        {
+            _output = output;
+            _configuration = new StagingEndToEndConfiguration();
+        }
+
+        [Fact]
+        public async Task VerifyCommitScenario()
+        {
+            var client = new StagingClient(new Uri(_configuration.StagingUri), _output);
+
+            // Create stage
+            var createStageResult = await VerifyCreateStage(client);
+
+            string stageId = createStageResult["Id"].ToString();
+
+            // Push packages
+            var pushedPackages = await VerifyPushPackages(client, stageId, _configuration.PackagesToPushCount);
+
+            // Commit stage
+            await VerifyCommitStage(client, stageId, pushedPackages);
+        }
+
+        [Fact]
+        public async Task VerifyStageManagement()
+        {
+            var client = new StagingClient(new Uri(_configuration.StagingUri), _output);
+
+            // Create stages
+            int stagesCount = 2;
+            var createdStageResults = new List<JObject>();
+
+            for (int i = 0; i < stagesCount; i++)
+            {
+                var createdStageResult = await VerifyCreateStage(client);
+                createdStageResults.Add(createdStageResult);
+            }
+
+            // Verify list user stages
+            await VerifyListUserStages(client, stagesCount, createdStageResults);
+
+            // Drop the stages
+            for (int i = 0; i < stagesCount; i++)
+            {
+                await VerifyDropStage(client, createdStageResults[i]["Id"].ToString());
+            }
+        }
+
+        private async Task VerifyDropStage(StagingClient client, string stageId)
+        {
+            _output.WriteLine($"Droping stage {stageId}");
+
+            var dropResult = await client.DropStage(stageId, _configuration.ApiKey);
+
+            dropResult["Id"].ShouldAllBeEquivalentTo(stageId, "Stage Id should be correct");
+            dropResult["Status"].ShouldAllBeEquivalentTo("Deleted");
+        }
+
+        private async Task VerifyListUserStages(StagingClient client, int stagesCount, List<JObject> createdStageResults)
+        {
+            var listUserStagesResult = await client.ListUserStages(_configuration.ApiKey);
+
+            foreach (var createdStageResult in createdStageResults)
+            {
+                var listUserStage = listUserStagesResult
+                    .First(x => x["Id"].ToString() == createdStageResult["Id"].ToString());
+
+                listUserStage.Should().NotBeNull($"Failed to find a list stage for stage {createdStageResult["Id"]}");
+                listUserStage["Status"].ToString().ShouldBeEquivalentTo("Active");

```

Comment:
done

---
In reply to: [64174159](https://github.com/NuGet/NuGet.Staging/pull/49#discussion_r64174159) [](ancestors = 64174159)